### ISSUE TITLE
Deletion of Ingress resource with ssl-passthrough=true causes TLS configuration to be lost

### ISSUE THREAD

**Issue by sivanov-nuodb** [haproxytech/kubernetes-ingress#544]
## Description

Deletion of all Ingress resources that have annotation `ingress.kubernetes.io/ssl-passthrough: "true"`, causes the configured TLS key needed to terminate HTTPS connections for all non-ssl-passthrough resources to be removed from `haproxy.cfg`, causing all HTTPS requests to be downgraded to HTTP.

Very similar to https://github.com/haproxytech/kubernetes-ingress/issues/475.

## Reproduction case
Install `haproxytech/kubernetes-ingress` with Helm chart version 1.30.5 and HAProxy version 1.10.2.

Create a Kubernetes service and Ingress resource that has ingress.kubernetes.io/ssl-passthrough: "true":

```sh
kubectl create service clusterip test --tcp=1234
kubectl apply -f - <<EOF
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: test
  annotations:
    ingress.kubernetes.io/ssl-passthrough: "true"
spec:
  rules:
  - host: "test.local"
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: test
            port:
              number: 1234
EOF
```

Wait for the HAProxy config to be updated and capture `haproxy.cfg` content:

```sh
kubectl exec "$HAPROXY_POD" -- cat /etc/haproxy/haproxy.cfg > haproxy-before.cfg
```

Delete the previously created Ingress resource;

```sh
kubectl delete ingress test
```

Wait about 10 seconds and capture haproxy.cfg again:
kubectl exec "$HAPROXY_POD" -- cat /etc/haproxy/haproxy.cfg > haproxy-after.cfg

If there are any Ingress resources that define L7 rules (i.e. not having ingress.kubernetes.io/ssl-passthrough: "true"), they will not longer work with HTTPS.
Comparing haproxy-before.cfg and haproxy-after.cfg reveals the problem. The TLS configuration is no longer present in the bind entries of frontend https:

```
$ diff -u haproxy-before.cfg  
--- haproxy-before.cfg	2023-06-12 13:04:01.000000000 +0300
+++ haproxy-after.cfg	2023-06-12 13:11:09.000000000 +0300
@@ -1,4 +1,4 @@
-# _version=64
+# _version=73
 # HAProxy Technologies
 # https://www.haproxy.com/
 # this file is not meant to be changed directly
@@ -60,8 +60,8 @@

 frontend https
   mode http
-  bind 0.0.0.0:8443 name v4 crt /etc/haproxy/certs/frontend ssl alpn h2,http/1.1
-  bind :::8443 name v6 crt /etc/haproxy/certs/frontend ssl v4v6 alpn h2,http/1.1
+  bind 0.0.0.0:8443 name v4
+  bind :::8443 name v6 v4v6
   http-request set-var(txn.base) base
   http-request set-var(txn.path) path
   http-request set-var(txn.host) req.hdr(Host),field(1,:),lower
```

## Expected behavior

The TLS configuration for the HTTPS `frontend` shouldn't be lost.

## Environment

Kubernetes: v1.24.14-eks-c12679a
haproxytech/kubernetes-ingress: 1.30.5
haproxytech/kubernetes-ingress: 1.10.2

**Comment by ivanmatmati** [comment]
Hi @sivanov-nuodb , thanks for reporting. I'll have a look at this.

**Comment by ivanmatmati** [comment]
I could recreate the issue. I'll fix it and let you know when it's available.

**Comment by ivanmatmati** [comment]
Hi @sivanov-nuodb , you should try the nightly build of the docker image. The fix has been introduced in commit 6f98c05e. To use the nightly build, simply replace your image version or latest tag with nightly.

**Comment by sivanov-nuodb** [comment]
@ivanmatmati I've tested with `HAProxy Ingress Controller v1.10.0 402c43c.dev.dirty` which works as expected.
Many thanks for fixing this! I expect that it will land in v1.10.5?
